wreck: fix event payload and memory leak in job.submit-nocreate

grondo opened the pull request:
This fixes #1491 by caching  the `struct wreck_job` created in `job.create` in the `active_jobs` hash, so it can be fetched and reused in `job.submit-nocreate`. The `job.submit-nocreate` rpc now only requires the `jobid` member be sent along in the payload.

In order to test the `submit` and `submit-nocreate` handlers better, a dummy sched module is built for the testsuite under `wreck/sched-dummy.la` and is used by a new test `t2000-wreck-dummy-sched.t` to test that the `submitted` events for `job.submit` and `job.submit-nocreate` are properly formed.

grondo commented:
One caveat with this fix-- `job.submit-nocreate` probably must be called on the same rank from which `job.create` is called. This is always the case with `flux-wreckrun`, but if other use cases somehow violate this assumption then `submit-nocreate` may not work. (actually it will suffer from a similar problem that this PR fixed)

garlick commented:
Might be good to pull in b70b76ee to this PR, which would allow you to declare:
```
MOD_NAME ("sched-dummy");
MOD_SERVICE ("sched");
```
and still answer the job module's ping request.

The module could then be loaded with `flux module load sched-dummy` rather than having to specify the full path.

Also, perhaps now would be a good time to insert a comment block in the `job_submit_only()` function describing how it is used by wreckrun.  I keep forgetting why we need it.

Sorry about that future leak!

grondo commented:
> wouldn't that require prepending to FLUX_MODULE_PATH anyway?
>
> I assumed not since the other wreck modules in the same directory can be
> loaded though I didn't check in detail.
>

That silly sched-dummy module is under `t/wreck/`, not a standard module
path from the build tree I don't think. I'll try it and see if it works
though.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/flux-framework/flux-core/pull/1492#issuecomment-385270933>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAtSUt4b-_Tvu-gd9ElF8AxJVBc4kHfYks5ttgVGgaJpZM4TrdG0>
> .
>

garlick commented:
Oh oops, probably not then.  Never mind!

grondo commented:
@garlick, I added a comment block as suggested to the top of job_submit_only. (also removed some erroneous cut-and-paste comment in the sched-dummy module)

garlick commented:
Looks good!  Thanks!

grondo commented:
Thanks!
